fv3core/stencils/d_sw.py

```diff
@@ -426,51 +392,42 @@ def max_d2_bg1():
     return max(spec.namelist.d2_bg, spec.namelist.d2_bg_k2)
 
 
-def get_column_namelist():
-    ks = [k[0] for k in k_bounds()]
-    col = {}
-    for ki in ks:
-        col[ki] = column_namelist_options(ki)
-    return col
-
-
-def get_single_column(key):
-    col = []
-    for k in range(0, grid().npz):
-        col.append(column_namelist_options(k)[key])
+def get_single_column(column_namelist, key):
+    col = [column_namelist[key][-1]] * grid().npz
+    for k in range(3):
+        col[k] = column_namelist[key][k]
     col.append(0.0)
     return col
 
 
-def column_namelist_options(k):
+def get_column_namelist():
```

Comment:
Can you add a docstring for this? Knowing what a column namelist is and why it exists would make it a lot more likely I can understand what this code is doing and why.